pkg/server/cluster_server.go

```diff
@@ -129,6 +140,17 @@ func (cs *clusterServer) GetConfig(cr poolRequest) (*runtime.RawExtension, error
 		return nil, fmt.Errorf("could not get controllerconfig: %w", err)
 	}
 
+	// we cannot mock this in a test env
+	if cs.configMapLister != nil {
+		cm, err := cs.configMapLister.ConfigMaps(ctrlcommon.MCONamespace).Get("kubeconfig-data")
+		if err != nil {
+			klog.Errorf("Could not get kubeconfig data: %v", err)
+		} else {
+			kc := cm.BinaryData["kubeconfig"]
+			addDataAndMaybeAppendToIgnition("/etc/kubernetes/kubeconfig", kc, &ignConf)
```

Comment:
As you have it now, once the configmap gets created, there will actually be 2 instances of `/etc/kubernetes/kubeconfig` referenced in the file which I think will break ignition?

You can probably make it conditional on the configmap existing and merge with existing append logic: if configmap exists, use it, if not, use bootstrap